### PR TITLE
#161967087: Users can signup/register to Authors Haven

### PR DESCRIPTION
**Pull request by Bryan-Cee**
**What does this PR do?**
-  Enable users to create an account.

**Description of tasks to be completed?**
- Create an account
- Send activation link to user via email

**How should this be manually tested?**

- Clone the repo and cd into the project folder
- Switch to the appropriate branch with `gco ft-signup-register-161967087`
- Run `npm install` to update your packages.
- Fire up the server with `npm start`
- Navigate to `http://localhost:3000/signup`

**What are the relevant pivotal tracker stories?**
- [#161967087](https://www.pivotaltracker.com/story/show/161967087)

**Screenshots**
screenshot 2019-01-17 at 17 48 17
screenshot 2019-01-17 at 17 49 17

### CODE REVIEW

File: src/actions/RegistrationActions.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+export const registerFailure = errors => ({
+  type: REGISTER_FAILURE,
+  isFetching: false,
+  errors
+});
+
+const registrationAsync = user => (dispatch) => {
+  dispatch(registerRequest());
+  instance
```
Comment (bettblake08) [inline src/actions/RegistrationActions.js]: Can you move the API call to an API class method, then dispatch actions according to what has been returned from API class method?

File: src/components/app.js [hunk 2]
```diff
@@ -1,62 +1,25 @@
...
-    };
-  }
-
-  componentDidMount() {
-    const { actions } = this.props;
-    const { welcomeMessage } = this.state;
-    actions.welcomeAction(welcomeMessage);
-    ArticleAPI.getArticles();
-  }
+  componentWillMount() {}
```
Comment (bettblake08) [inline src/components/app.js]: I believe you should remove this. This lifecycle method is deprecated.

File: src/reducers/index.js [hunk 3]
```diff
@@ -1,8 +1,9 @@
 import { combineReducers } from "redux";
 import { connectRouter } from "connected-react-router";
-import welcomeReducer from "./welcomeReducer";
+
+import registration from "./registrationReducer";
 
 export default history => combineReducers({
   router: connectRouter(history),
-  welcomeReducer
+  registration
```
Comment (bettblake08) [inline src/reducers/index.js]: Add the suffix Reducer to the registration reducer.

**Review by bettblake08 — CHANGES_REQUESTED** [review]

**Review by denomash — APPROVED** [review]

**Review by davishooly — APPROVED** [review]

**Review by bettblake08 — APPROVED** [review]
👍

**Review by Darius-Ndubi — APPROVED** [review]
👍

File: src/actions/registrationActions.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import toastr from "../helpers/toastrConfig";
+
+import { REGISTER_REQUEST, REGISTER_SUCCESS, REGISTER_FAILURE } from "./actionTypes";
+import instance from "../api/axiosConfig";
+
+/* eslint-disable import/prefer-default-export */
+export const registerRequest = () => ({
+  type: REGISTER_REQUEST,
+  isFetching: true
```
Comment (justmesam) [inline src/actions/registrationActions.js]: you don't have to have this as part of the action  you only need to change it on the reducer

File: src/assets/style/main.scss [hunk 5]
```diff
@@ -1,6 +1,20 @@
...
-}
\ No newline at end of file
+}
+
+.modal {
+  text-align: right !important;
+}
+
+.ui.active.transition.visible.dimmer,
+.ui.page.modals.dimmer.transition.visible.active {
```
Comment (justmesam) [inline src/assets/style/main.scss:12]: which class is this and wouldn't have been easier to create a new reference class ?

File: src/assets/style/main.scss [hunk 6]
```diff
@@ -1,6 +1,20 @@
...
-.text
-{
+.text {
   color: $color-green;
-}
\ No newline at end of file
+}
+
+.modal {
+  text-align: right !important;
```
Comment (justmesam) [inline src/assets/style/main.scss:8]: refrain from using important.

File: src/components/containers/registration.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+      switch (name) {
+        case "username":
+          /* eslint-disable no-case-declarations */
+          /* eslint-disable react/no-access-state-in-setstate */
+          /* eslint-disable no-useless-escape */
+          const usernameReg = /^[a-zA-Z][a-zA-Z0-9]{3,}$/;
+
+          if (value.match(usernameReg) === null) {
+            this.setState({
+              ...this.state,
```
Comment (justmesam) [inline src/components/containers/registration.js]: why are you spreading the state here??

File: src/assets/style/main.scss (hunk 5 above)
Comment (Bryan-Cee) [reply src/assets/style/main.scss:12]: Both classes are inaccessible directly, they are generated by semantic UI React components, that is why I chose to pick the whole class as is to effect the change that was needed.

File: src/assets/style/main.scss (hunk 6 above)
Comment (Bryan-Cee) [reply src/assets/style/main.scss:8]: I needed to override the default text-align property set by semantic UI React components, is there any other way to have it done without using `! important` ?

Comment (justmesam) [reply src/assets/style/main.scss:8]: you can unset the value

**Pull request merged**